### ISSUE TITLE
Alt-text for emojis

### ISSUE THREAD

**Issue by sol-foxie** [glitch-soc/mastodon#720]
I noticed that there is syntax for writing emojis in the form of `:smile:` which is then converted to emoji form. (I presume the conversion is done on the frontend?) I was thinking, especially with the existence of custom emoji, it might be useful to be able to specify a readable version of the emoji.

For example, what if typing :blobcatcoffee:[takes a long sip of coffee] would cause the screen-reader to read out "takes a long sip of coffee", while only the blobcatcoffee glyph is displayed on the page itself. Additionally, we could expose this information to non-screen reader users through hover text or something like that.

What do y'all think? Hopefully I articulated my point well enough. I'm new to Mastodon, but not new to coding, and if you think it's a good idea I might go ahead and look at ways to implement this.

**Comment by ThibG** [comment]
I think specifying the alt text when the emoji is used is inconvenient and it's probably going to be very hard to federate in a meaningful way. I'm more interested in adding metadata to the emoji itself, and that could be part of it, although it may require a bit of thinking design-wise, and I'd have to check what already exists in the protocol to not reinvent the wheel.

**Comment by sol-foxie** [comment]
I agree it’s kinda clunky, but at the same time could be useful/fun for injecting added semantics. E.g. `:information_desk_person:[sassy hair flip]` vs `:information_desk_person:[and that’s the tea]`. I imagine we’d just federate the plain text and let frontends interpret it as they will. That’s part of the reason why I suggested it as a glitch-soc feature, it might make sense for other forks to have their own way of displaying the text.

However, your idea is good too, and I’m willing to help implement it if that’s what you’d prefer. If that’s the case, let me know what you find re: existing stuff in the protocol.

**Comment by ThibG** [comment]
We could probably use the icon's `summary` field or make up a `summary` field for the `Emoji` object type. It probably should be upstream, though (especially since the `Emoji` type is namespaced there).

**Comment by nightpool** [comment]
right now on the activitypub level, emoji are considered to be shared across all statuses using them. it would take more-then-a-little protocol level wrangling to allow it to be specified per-status (not to mention different usages within the same text)

however, for adding metadata to an emoji for any given server, using the `as:summary` field would be the correct way to go (no need to make one up, since the relation already exists....)

**Comment by sol-foxie** [comment]
So I guess the question is, is it currently possible to set the ‘as:summary’ field in Mastodon? I admit I’ve never set up an instance so I don’t know the UI for setting custom emoji??

**Comment by BenLubar** [comment]
Don't set the alt attribute on the img tag to anything that won't create the emoji when pasted into the post editor, because that'll break copy-paste.
